### PR TITLE
- add support for full dump via cmd line.

### PR DESCRIPTION
**Pull request by igorshevach**
usage:
enable full dump creation:
-full-memory-crash-report=1
force disable full dump creation:
-full-memory-crash-report=0.

IMPORTANT:
vendor/breakpad should be updated to latest master version since commit 33a4375 does not contain required code fixes.

### CODE REVIEW

File: atom/common/crash_reporter/crash_reporter_win.cc [hunk 1]
```diff
@@ -35,10 +36,28 @@ namespace crash_reporter {
 
 namespace {
 
-// Minidump with stacks, PEB, TEB, and unloaded module list.
-const MINIDUMP_TYPE kSmallDumpType = static_cast<MINIDUMP_TYPE>(
-    MiniDumpWithProcessThreadData |  // Get PEB and TEB.
-    MiniDumpWithUnloadedModules);  // Get unloaded modules when available.
+
+auto getDumpType()
```
Comment (cifratila) [inline atom/common/crash_reporter/crash_reporter_win.cc]: Could you use spaces and not tabs for indentation?

File: atom/common/crash_reporter/crash_reporter_win.cc [hunk 2]
```diff
@@ -177,10 +196,9 @@ void CrashReporterWin::InitBreakpad(const std::string& product_name,
       MinidumpCallback,
       this,
       google_breakpad::ExceptionHandler::HANDLER_ALL,
-      kSmallDumpType,
+	  getDumpType(),
```
Comment (cifratila) [inline atom/common/crash_reporter/crash_reporter_win.cc]: Same here, use spaces.

File: atom/common/crash_reporter/crash_reporter_win.cc [hunk 3]
```diff
@@ -177,10 +196,9 @@ void CrashReporterWin::InitBreakpad(const std::string& product_name,
...
       pipe_name.c_str(),
-      GetCustomInfo(product_name, version, company_name, upload_to_server)));
-
+	  GetCustomInfo(product_name, version, company_name, upload_to_server)));
```
Comment (cifratila) [inline atom/common/crash_reporter/crash_reporter_win.cc]: Same here, use spaces.

File: atom/common/crash_reporter/crash_reporter_win.cc [hunk 4]
```diff
@@ -14,6 +14,7 @@
 #include "content/public/common/result_codes.h"
 #include "gin/public/debug.h"
 #include "sandbox/win/src/nt_internals.h"
+#include "base/command_line.h"
```
Comment (cifratila) [inline atom/common/crash_reporter/crash_reporter_win.cc]: Not sure if I am a big fan of adding extra command line dependency here, couldn't you use the extra parameters provided in crash_reporter.cc and depending on existence or not of one do the same logic?

**Comment by jkleinsc** [comment]
Closing this issue due to inactivity.  If someone wants to pick this issue up, please comment on this PR.

**Pull request closed without merging**